### PR TITLE
fix(rule): add 1-hour tolerance for composting cycle timeframe boundaries

### PR DESCRIPTION
**Pull request by sangalli**
## Summary

- Adds a 1-hour tolerance to the composting cycle timeframe rule's min/max boundary checks (60-180 days)
- Fixes 12 MassIDs failing with "59 days outside valid range" when the actual duration is ~59.9999 days
- Root cause: integrator timestamps have seconds of jitter around midnight (Drop-off at `00:00:08`, Recycled at `00:00:00`), and `differenceInDays` truncates to 59

## Changes

- **constants**: Extract `COMPOSTING_CYCLE_MIN_DAYS`, `COMPOSTING_CYCLE_MAX_DAYS`, `HOURS_PER_DAY`, and `TOLERANCE_IN_HOURS` constants
- **processor**: Use `differenceInHours` for boundary comparison with ±1h tolerance; keep `differenceInDays` for display
- **test-cases**: Add jitter tolerance scenarios for both min and max boundaries

## Validated against production data

```
MassID: 62c7d0ba-509d-435f-a533-203c119c466d
Drop-off: 2025-01-24T00:00:08-03:00
Recycled: 2025-03-25T00:00:01-03:00
Result:  ✓ PASSED (was FAILED)
```

## Test plan

- [x] All 10 unit tests pass (including 2 new tolerance scenarios)
- [x] 100% code coverage
- [x] Lint, typecheck pass
- [x] Verified against production MassID `62c7d0ba` via `run-rule` CLI

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Enhanced composting cycle validation with hour-level precision for improved accuracy.
  * Added 1-hour tolerance window to composting cycle boundaries (60-180 days) to account for minor timing variations.

* **Tests**
  * Expanded test coverage for composting cycle boundary condition validation.

### CODE REVIEW

File: libs/methodologies/bold/rule-processors/mass-id/composting-cycle-timeframe/src/composting-cycle-timeframe.constants.ts [hunk 1]
```diff
@@ -2,16 +2,27 @@ import { DocumentEventName } from '@carrot-fndn/shared/methodologies/bold/types'
...
+export const COMPOSTING_CYCLE_MAX_DAYS = 180;
+export const COMPOSTING_CYCLE_MIN_DAYS = 60;
+export const HOURS_PER_DAY = 24;
+
+/**
+ * Tolerance applied to the min/max boundary comparison to account for
+ * timestamp jitter from integrators (e.g. events logged a few seconds
+ * after midnight instead of exactly at midnight).
+ */
+export const TOLERANCE_IN_HOURS = 1;
```
Comment (cris-santos) [inline libs/methodologies/bold/rule-processors/mass-id/composting-cycle-timeframe/src/composting-cycle-timeframe.constants.ts:14]: we should also probably update the rule description in `libs/methodologies/bold/rule-processors/mass-id/composting-cycle-timeframe/src/composting-cycle-timeframe.rule-definition.ts`

**Review by cris-santos — APPROVED** [review]

**Review by AMarcosCastelo — APPROVED** [review]
